.spectral.yaml

```diff
@@ -1,2 +1,6 @@
 extends: '@rhoas/spectral-ruleset'
-
+rules:
+  no-$ref-siblings: off
+  rhoas-response-media-type: off
+  rhoas-error-schema: off
+  oas3-valid-schema-example: off
```

Comment:
What's the consequence of turning these off?

Reply:
our rhoas-error-schema appears to be broken anyway so as far as I can see every API this has ran against has failed. 

The oas3-valid-schema-example appears to error our when the value needed is actually included but is nested or the value is included in the doc but one must navigate to it via the $ref. The rule isn't capable of reading nested items or $refs so I proposed in the google doc I attached that we disable this for now and write a custom rule when we upgrade to spectral v6 (which I'll be doing shortly anyway) to check for these scenarios. This happens quite frequently throughout the doc

Reply:
Thank you That makes sense. Do you mind adding in comment for now, briefly saying why it is disabled and when it can be enabled?

Reply:
The error schema may have to be rethought. This is why I was asking about the errors earlier. I'm not looking to make massive changes for the sake of the neighbors. Especially breaking changes, I'll be looking into this further as we upgrade to v6 and look to maybe create some new rules but edit some that are currently unsatisfactory

Reply:
> The error schema may have to be rethought. 

I think just changing the way the schema is declared should be fine in my opinion as long as we do not change or remove the properties. e.g a change like below to the schema should be okay.
```yaml
 Error:
      type: object
      required: [id, kind, href, code, reason]
      properties:
        id:
         type: string
        kind:
         type: string
        code:
         type: string
        reason:
         type: string
        href:
         type: string
        operation_id:
         type: string
```

Reply:
I left a detailed reply here, has it just vanished or did you get to see it @machi1990 ?

Reply:
I have tried that and it hasn't worked unfortunately

> > > > The file now passes linter.
> > > 
> > > 
> > > Awesome. Do you think we are at a point where we could enabled spectral validation as part of CI workflow?
> > > so long as it passes the tests yes that was my next step to enable on CI
> > 
> > 
> > @machi1990
> > I do not see any errors when I run that command. I did initially until I replaced 'url' with 'email'
> > > Can you run `make openapi/generate code/fix` and commit the changes as well. That will enable the full CI run and we'll see how much of the generated code are usable as it. Seeing the changes, I do not expect that the DTOs have changed much.
> > 
> > 
> > it still appears to fail CI though, although I'm not sure it ran again after force push. wasn't there a bot command to rerun checks?
> 
> So CI fails because the generated files have not been committed and force pushed.

so the files like this:
`internal/kafka/internal/api/public/model_cloud_region.go` 
which are generated after `make openapi/generate code/fix` is run
need to be committed too? that does make sense

Reply:
Yes.